azext_iot/operations/hub.py

```diff
@@ -181,6 +202,127 @@ def iot_device_delete(cmd, device_id, hub_name=None, resource_group_name=None, l
+    if remove_all:
+        result = _iot_device_children_list(cmd, device_id, hub_name, resource_group_name, login)
+        if not result:
+            raise CLIError('No registered child devices found for "{}" edge device.'.format(device_id))
+        for non_edge_device_id in ([str(x['deviceId']) for x in result]):
+            nonedge_device = _iot_device_show(target, non_edge_device_id.strip())
+            devices.append(nonedge_device)
+    elif child_list:
+        edge_device = _iot_device_show(target, device_id)
+        _validate_edge_device(edge_device)
+        for non_edge_device_id in child_list.split(','):
+            nonedge_device = _iot_device_show(target, non_edge_device_id.strip())
+            _validate_nonedge_device(nonedge_device)
+            _validate_child_device(nonedge_device)
+            if nonedge_device['deviceScope'] != edge_device['deviceScope']:
+                raise CLIError('The entered child device "{}" isn\'t assigned as a child of edge device "{}"'
+                               .format(non_edge_device_id.strip(), device_id))
+            if nonedge_device['deviceScope'] == edge_device['deviceScope']:
+                devices.append(nonedge_device)
+                continue
+
+    for device in devices:
+        _update_nonedge_devicescope(target, device)
+
+
+def iot_device_children_list(cmd, device_id, hub_name=None, resource_group_name=None, login=None):
+    result = _iot_device_children_list(cmd, device_id, hub_name, resource_group_name, login)
+    if not result:
+        raise CLIError('No registered child devices found for "{}" edge device.'.format(device_id))
+    return ', '.join([str(x['deviceId']) for x in result])
+
+
+def _iot_device_children_list(cmd, device_id, hub_name=None, resource_group_name=None, login=None):
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    device = _iot_device_show(target, device_id)
+    _validate_edge_device(device)
+    query = ('select * from devices where capabilities.iotEdge=false and deviceScope=\'{}\''
+             .format(device['deviceScope']))
+    return iot_query(cmd, query, hub_name, None, resource_group_name, login=login)
+
+
+def _update_nonedge_devicescope(target, nonedge_device, deviceScope=''):
+    service_sdk, errors = _bind_sdk(target, SdkType.service_sdk)
+    try:
+        nonedge_device['deviceScope'] = deviceScope
+        etag = nonedge_device.get('etag', None)
+        if etag:
+            headers = {}
+            headers["If-Match"] = '"{}"'.format(etag)
+            service_sdk.create_or_update_device(nonedge_device['deviceId'], nonedge_device, custom_headers=headers)
+        else:
+            raise LookupError("device etag not found.")
+    except errors.CloudError as e:
+        raise CLIError(unpack_msrest_error(e))
+
+
+def _validate_edge_device(device):
+    if not device['capabilities']['iotEdge']:
+        raise CLIError('The device should be edge device.')
```

Comment:
It certainly helps user experience when you add additional detail to the error, in this instance, addition text to call out the device Id failing this check (like you have in _validate_nonedge_device).

Reply:
Fixed